docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,428 @@
+- [Backing up current cluster data](#backup)
+- [Patch the cluster to use the new OIDC](#patch-cluster)
+- [Revoke public access to S3 Bucket](#revoke-s3-public-access)
+- [Rollback to OIDC with S3 Public URL](#rollback)
+- [Rollback to OIDC with S3 Public URL](#delete)
+
+## Prerequisites <a name="Prerequisites"></a>
+
+- An OCP Cluster created on AWS with manual authentication mode with STS using S3 Bucket as OIDC URL (issuerURL)
+
+
+- Environment variables exported:
+```bash
+export CLUSTER_NAME="<ChangeMe>"
+export OIDC_BUCKET_HOST=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}')
+export OIDC_BUCKET_NAME=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$1}')
+export CLUSTER_REGION=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$3}')
+```
+
+- Make sure you can reach (read) the S3 Bucket created with the default name
+
+```bash
+aws s3 ls --region $CLUSTER_REGION s3://$OIDC_BUCKET_NAME
+```
+
+- A clean work directory: a lot of files will be created, make sure you switched to a new work directory to save the files properly (it can be used in the future for rollback)
+
+## Test the existing token <a name="test-token"></a>
+
+This section is to make sure everything is working correctly in your existing environment.
+
+The steps described below will use the credentials provided to the `machine-api` component, trying to assume the role using `aws-cli`.
+
+It's expected that the existing token will be able to authenticate in AWS. Otherwise, you must abort the operations and do not try to run any step described in the next sections
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^web_identity_token_file" |\
+    awk '{print$3}')
+
+# Get Controler's pod
+CAPI_POD=$(oc get pods -n openshift-machine-api \
+    -l api=clusterapi \
+    -o jsonpath='{.items[*].metadata.name}')
+
+# Extract tokens from the pod
+TOKEN=$(oc exec -n openshift-machine-api ${CAPI_POD} \
+    -c machine-controller -- cat ${TOKEN_PATH})
+
+echo $TOKEN | awk -F. '{ print $2 }' | base64 -d 2>/dev/null | jq .iss
+
+IAM_ROLE=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^role_arn" |\
+    awk '{print$3}')
+
+echo $IAM_ROLE
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}"
+```
+
+## Setup new OIDC with CloudFront Distribution <a name="setup"></a>
+
+### Create the CloudFront Distribution <a name="setup-cloudfront"></a>
+
+- Create the Origin Access Identity (OAI)
```

Comment:
OAI now is a legacy mode! We can restrict directly from CloudFront Arn, for example of S3 Bucket policy:

```json
{
        "Version": "2008-10-17",
        "Id": "PolicyForCloudFrontPrivateContent",
        "Statement": [
            {
                "Sid": "AllowCloudFrontServicePrincipal",
                "Effect": "Allow",
                "Principal": {
                    "Service": "cloudfront.amazonaws.com"
                },
                "Action": "s3:GetObject",
                "Resource": "arn:aws:s3:::oidc-demo.devcluster.openshift.com/*",
                "Condition": {
                    "StringEquals": {
                      "AWS:SourceArn": "arn:aws:cloudfront::[redacted]:distribution/[redacted:CfnDistributionArn]"
                    }
                }
            }
        ]
      }
```

Reply:
we should replace the steps referencing OAI fixing the bucket policy with the recommended approach.

Reply:
https://aws.amazon.com/pt/blogs/networking-and-content-delivery/amazon-cloudfront-introduces-origin-access-control-oac/

Reply:
OAC is an alternative recently added. This document is not covering right know due the priority of this PR. We must consider this in the near future both for ccoctl and docs/template updates

Reply:
https://issues.redhat.com/browse/CCO-347 is the card!